<a href="https://colab.research.google.com/github/miajags/EMSC2010-W9-global-modelling/blob/main/EMSC2010_W9_L1_NB1_globe_temperature_modelling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# EMSC2010-W9-L1-NB1
Model Earth's surface temperature (in Kelvin) using a zero-dimensional energy balance model without an atmosphere.

$T_e = \left[ \frac{S}{4 \alpha} (1-A)\right]^{1/4}$

where:
* $T_e$ is Earth's surface temperature in Kelvin
* $S$ is the solar flux (1366 W/m$^2$)
* $A$ is albedo (currently around 0.3)
* $\alpha$ is the Stefan-Boltmann constant (5.67E-8 W/m$^2$/K$^4$)

In [2]:
#define values

S = 1366 #Solar flux
A = 0.3 #Albedo
a = 5.67E-8 #Stefan-Boltmann constant

In [4]:
#turn the equation into computer code

T_e = ((S/(4*a))*(1-A))**(1/4)
print(T_e)

254.81584054796193


254.8 Kelvin is Earth's surface temperature considering our values.

But, this is inconsistent with what we expect Earth's average surface temperature to be - much too cold!
254.8 Kelvin is about -20 degrees celsius


But... we assumed there was no atmosphere (not true)
The atmospher traps heat, warms temeperature... greenhouse effect!

In [5]:
#let's change the values

B = 0 #alternative albedo

In [6]:
#turn the equation into computer code

T_e2 = ((S/(4*a))*(1-B))**(1/4)
print(T_e2)

278.5812689476053


Planet got colder by about 20 Kelvin when albedo is 0 rather than 0.3.

In [9]:
#Hail mary example, solar flare dimmed by 10%

Hail_mary = S*0.9
print(Hail_mary)

1229.4


In [10]:
#turn the equation into computer code

T_e3 = ((Hail_mary/(4*a))*(1-A))**(1/4)
print(T_e3)

248.19158334222595


Planet got colder by about 10 Kelvin when albedo is 0.3 and the solar flare is reduced by 10%.